### PR TITLE
[thci] retrieve BA port from mDNS response

### PR DESCRIPTION
**Pull request by edmont**
Modify `mdns_query` to accept a unicast destination and parse the mDSN response to find the Border Router's Border Agent port.

### CODE REVIEW

File: tools/harness-thci/OpenThread.py [hunk 1]
```diff
@@ -2322,10 +2322,6 @@ def startNativeCommissioner(self, strPSKc='GRLPASSPHRASE'):
         print(cmd)
         return self.__executeCommand(cmd)[-1] == 'Done'
 
-    @API
-    def getBorderAgentPort(self):
```
Comment (simonlingoogle) [inline tools/harness-thci/OpenThread.py:2326]: We don't need this API anymore?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: tools/harness-thci/OpenThread.py (hunk 1 above)
Comment (edmont) [reply tools/harness-thci/OpenThread.py:2326]: We don't since it's always possible to inquiry a Border Router for this parameter via mDNS.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**